src/js/Sidebar.js

```diff
@@ -0,0 +1,120 @@
+import { getUserName } from "./api.js";
+import { push } from "./router.js";
+import { getAllDocuments, addNewDocument, deleteDocument } from "../utils/Documents.js";
+import { getItem, setItem } from "../utils/storage.js";
+
+export default function Sidebar({ target, initialState = {} }) {
+    if (!(this instanceof Sidebar)) {
+        return new Sidebar({ target, initialState });
+    }
+
+    // *****************************
+    // * Valuables and Functions   *
+    // *****************************
+
+    const sidebar = document.createElement("nav");
+    const username = getUserName();
+    const documents = () => getItem(`username_${username}`, []);
+
+    this.state = initialState;
+
+    // *****************************
+    // * Setting attribute         *
+    // *****************************
+
+    sidebar.setAttribute("class", "sidebar");
+
+    sidebar.innerHTML = `
+                        <header id="username">${username}</header>
+                        <ul id="treeUL">${documents()
+                            .map((root) => getAllDocuments(root, parseInt(this.state.documentId)))
+                            .join("")}
+                        </ul>
+                        <ul>
+                            <button type="button" class="add-btn">Add Document</button>
+                        </ul>
+                        `;
+
+    // *****************************
+    // * Function for Rendering    *
+    // *****************************
+
+    target.appendChild(sidebar);
+
+    const toggleMouseOver = (e) => {
+        const target = e.target;
+
+        if (target.className.includes("document") || target.className.includes("childDocument")) {
+            target.classList.toggle("mouse-over");
+        }
+    };
+
+    this.setState = (nextState) => {
+        this.state = nextState;
+        this.render();
+    };
+
+    this.render = () => {
+        const { documentId } = this.state;
+        const ul = sidebar.querySelector("#treeUL");
+
+        if (ul) {
+            ul.innerHTML = `${documents()
+                .map((root) => getAllDocuments(root, parseInt(documentId)))
+                .join("")}`;
+        }
+    };
+
+    // Click Event
+    sidebar.addEventListener("click", async (e) => {
+        const target = e.target;
+
+        if (target) {
+            const documentId = target.dataset.id;
+
+            if (target.className.includes("document")) {
```

Comment:
```suggestion
            if (target.className === "document") {
```
삭제, 추가버튼이 작동하고 있지 않은데, Create, Delete 버튼에도 document가 들어가기 때문에 else if문이 작동하지 않고, 그것 때문에 오류가 발생하고 있는 것 같아요!

Reply:
버그가 많습니다 하하하
얼른 고치겠습니다 :)
감사합니다~~